### PR TITLE
Feature/ht 77/refresh user's token when access token is expiring

### PR DESCRIPTION
**Pull request by kalibria**
If the token is about to expire, send a request to token/refresh to refresh the token

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "react-scripts test jest",
```
Comment (ikushlianski) [inline package.json]: I don't think we need this `jest` bit. React-scripts automatically uses jest. Pls remove

File: src/auth/authenticationManager.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const { isTokenExpired } = require('./authenticationManager');
+
+describe('testing isTokenExpired function', () => {
```
Comment (ikushlianski) [inline src/auth/authenticationManager.test.js]:
```suggestion
describe('isTokenExpired', () => {
```

File: src/auth/authenticationManager.ts [hunk 3]
```diff
@@ -100,3 +104,61 @@ export const useCheckIsLoggedIn = () => {
...
+      })
+      .join(''),
+  );
+
+  const { exp } = JSON.parse(jsonPayload);
+  return exp;
+}
+
+export const isTokenExpired = (decodeToken: number) => {
+  const timeMsUntilTokenExpires = 5 * 60 * 1000;
```
Comment (ikushlianski) [inline src/auth/authenticationManager.ts:125]: great that you mentioned `Ms` in the variable name!

File: src/auth/authenticationManager.ts [hunk 4]
```diff
@@ -100,3 +104,61 @@ export const useCheckIsLoggedIn = () => {
...
+        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+      })
+      .join(''),
+  );
+
+  const { exp } = JSON.parse(jsonPayload);
+  return exp;
+}
+
+export const isTokenExpired = (decodeToken: number) => {
```
Comment (ikushlianski) [inline src/auth/authenticationManager.ts]:
```suggestion
export const isTokenExpired = (tokenExpirationTimestamp: number) => {
```

Comment (ikushlianski) [reply src/auth/authenticationManager.ts]: Just not clear what `decodeToken` is

File: src/auth/authenticationManager.ts [hunk 5]
```diff
@@ -100,3 +104,61 @@ export const useCheckIsLoggedIn = () => {
...
+  const timeMsUntilTokenExpires = 5 * 60 * 1000;
+  const isTimeUp = decodeToken * 1000 - timeMsUntilTokenExpires;
+  const expired = Date.now() >= decodeToken * 1000 || Date.now() >= isTimeUp;
+
+  return expired;
+};
+
+export function useIsIdTokenExpired() {
+  let currentIdToken = sessionStorage.getItem('IdToken');
```
Comment (ikushlianski) [inline src/auth/authenticationManager.ts]: use `const` if your variable is not mutated

File: package.json (hunk 1 above)
Comment (kalibria) [reply package.json]: removed

File: src/auth/authenticationManager.test.js (hunk 2 above)
Comment (kalibria) [reply src/auth/authenticationManager.test.js]: changed

File: src/auth/authenticationManager.ts (hunk 4 above)
Comment (kalibria) [reply src/auth/authenticationManager.ts]: renamed

File: src/auth/authenticationManager.ts (hunk 5 above)
Comment (kalibria) [reply src/auth/authenticationManager.ts]: changed

**Pull request merged**